editor/export/export_template_manager.cpp

```diff
@@ -655,39 +656,80 @@ void ExportTemplateManager::_hide_dialog() {
 	hide();
 }
 
-bool ExportTemplateManager::can_install_android_template() {
+String ExportTemplateManager::get_android_build_directory(int p_export_preset_index) {
+	Ref<EditorExportPreset> export_preset = p_export_preset_index >= 0 ? EditorExport::get_singleton()->get_export_preset(p_export_preset_index) : Ref<EditorExportPreset>();
+	if (export_preset.is_valid()) {
+		String gradle_build_dir = export_preset->get("gradle_build/gradle_build_directory");
+		if (!gradle_build_dir.is_empty()) {
+			return gradle_build_dir.path_join("build");
+		}
+	}
+	return "res://android/build";
+}
+
+String ExportTemplateManager::get_android_source_zip(int p_export_preset_index) {
+	Ref<EditorExportPreset> export_preset = p_export_preset_index >= 0 ? EditorExport::get_singleton()->get_export_preset(p_export_preset_index) : Ref<EditorExportPreset>();
+	if (export_preset.is_valid()) {
+		String android_source_zip = export_preset->get("gradle_build/android_source_template");
+		if (!android_source_zip.is_empty()) {
+			return android_source_zip;
+		}
+	}
+
 	const String templates_dir = EditorPaths::get_singleton()->get_export_templates_dir().path_join(VERSION_FULL_CONFIG);
-	return FileAccess::exists(templates_dir.path_join("android_source.zip"));
+	return templates_dir.path_join("android_source.zip");
 }
 
-Error ExportTemplateManager::install_android_template() {
-	const String &templates_path = EditorPaths::get_singleton()->get_export_templates_dir().path_join(VERSION_FULL_CONFIG);
-	const String &source_zip = templates_path.path_join("android_source.zip");
+String ExportTemplateManager::get_android_template_version(int p_export_preset_index) {
```

Comment:
Instead of `get_android_template_version`, maybe we can use `get_android_template_identifier` and add a comment to indicate that the identifier is the version for the default template, and the full path for custom templates.

Reply:
Fixed!